src/gui/src/dbDescriptors.cpp

```diff
@@ -1821,4 +1836,355 @@ bool DbItermAccessPointDescriptor::getAllObjects(SelectionSet& objects) const
+  }
+  return Selected();
+}
+
+bool DbRegionDescriptor::lessThan(std::any l, std::any r) const
+{
+  auto l_layer = std::any_cast<odb::dbRegion*>(l);
+  auto r_layer = std::any_cast<odb::dbRegion*>(r);
+  return l_layer->getId() < r_layer->getId();
+}
+
+bool DbRegionDescriptor::getAllObjects(SelectionSet& objects) const
+{
+  auto* chip = db_->getChip();
+  if (chip == nullptr) {
+    return false;
+  }
+  auto* block = chip->getBlock();
+  if (block == nullptr) {
+    return false;
+  }
+
+  for (auto* region : block->getRegions()) {
+    objects.insert(makeSelected(region, nullptr));
+  }
+  return true;
+}
+
+//////////////////////////////////////////////////
+
+DbModuleDescriptor::DbModuleDescriptor(odb::dbDatabase* db) :
+    db_(db)
+{
+}
+
+std::string DbModuleDescriptor::getShortName(std::any object) const
+{
+  auto* module = std::any_cast<odb::dbModule*>(object);
+  return module->getName();
+}
+
+std::string DbModuleDescriptor::getName(std::any object) const
+{
+  auto* module = std::any_cast<odb::dbModule*>(object);
+  return module->getHierarchicalName();
+}
+
+std::string DbModuleDescriptor::getTypeName() const
+{
+  return "Module";
+}
+
+bool DbModuleDescriptor::getBBox(std::any object, odb::Rect& bbox) const
+{
+  auto* module = std::any_cast<odb::dbModule*>(object);
+  auto insts = module->getInsts();
+  if (insts.empty()) {
+    return false;
+  }
+  bbox.mergeInit();
+  for (auto* inst : insts) {
+    auto* box = inst->getBBox();
+    odb::Rect box_rect;
+    box->getBox(box_rect);
+    bbox.merge(box_rect);
+  }
+  return true;
+}
+
+void DbModuleDescriptor::highlight(std::any object,
+                                  Painter& painter,
+                                  void* additional_data) const
+{
+  auto* module = std::any_cast<odb::dbModule*>(object);
+
+  auto* inst_descriptor = Gui::get()->getDescriptor<odb::dbInst*>();
+  for (auto* inst : module->getInsts()) {
+    inst_descriptor->highlight(inst, painter, nullptr);
+  }
+
+  for (auto* children : module->getChildren()) {
+    highlight(children->getMaster(), painter, nullptr);
+  }
+}
+
+Descriptor::Properties DbModuleDescriptor::getProperties(std::any object) const
+{
+  auto* module = std::any_cast<odb::dbModule*>(object);
+  auto* mod_inst = module->getModInst();
+
+  auto* gui = Gui::get();
+
+  Properties props;
+  if (mod_inst != nullptr) {
+    auto* parent = mod_inst->getParent();
+    if (parent != nullptr) {
+      props.push_back({"Parent", gui->makeSelected(parent)});
+    }
+
+    auto* group = mod_inst->getGroup();
+    if (group != nullptr) {
+      props.push_back({"Group", gui->makeSelected(group)});
+    }
+  }
+
+  SelectionSet children;
+  for (auto* child : module->getChildren()) {
+    children.insert(gui->makeSelected(child->getMaster()));
+  }
+  if (!children.empty()) {
+    props.push_back({"Children", children});
+  }
+
+  SelectionSet insts;
+  for (auto* inst : module->getLeafInsts()) {
```

Comment:
This should be getInsts() otherwise you are showing all instances recursive which makes it unclear which are local to this module.